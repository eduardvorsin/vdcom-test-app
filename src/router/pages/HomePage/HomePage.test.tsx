import React from 'react';

import { render, screen, waitFor } from '@testing-library/react';
import wrapWithRedux from '../../../tests/helpers/wrapWithRedux';
import addMemoryRouter from '../../../tests/helpers/addMemoryRouter';

describe('HomePage tests', () => {
  it('renders correctly', async () => {
    render(wrapWithRedux(addMemoryRouter(), {}));

    await waitFor(() => {
      expect(screen.getByRole<HTMLElement>('banner')).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByRole<HTMLButtonElement>('button', { name: /open navigation menu/i })).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByRole<HTMLInputElement>('textbox')).toBeInTheDocument();
    });
  });

  it('is a basic snapshot', async () => {
    render(wrapWithRedux(addMemoryRouter(), {}));

    expect(await screen.findByTestId<HTMLDivElement>('home-page')).toMatchSnapshot();
  });

  it('is a snapshot with a screen width greater than 900px', async () => {
    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(1000);
    render(wrapWithRedux(addMemoryRouter(), {}));

    expect(await screen.findByTestId<HTMLDivElement>('home-page')).toMatchSnapshot();
  });
});
