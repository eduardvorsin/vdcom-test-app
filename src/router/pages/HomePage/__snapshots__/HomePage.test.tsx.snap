// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests is a basic snapshot 1`] = `
<div
  class="MuiBox-root css-1svsvwq"
  data-testid="home-page"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-36h2sv-MuiPaper-root-MuiAppBar-root"
    data-testid="header"
  >
    <div
      class="MuiBox-root css-sdx3s9"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ubps8a-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        open navigation menu
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-112rk81-MuiTypography-root-MuiLink-root"
        href="/contacts"
      >
        <img
          alt="logo"
          class="css-12pywbb"
          src="assets"
        />
      </a>
      <form
        class="css-1pho46g"
        data-testid="search-form"
        novalidate=""
      >
        <div
          class="MuiBox-root css-19sk4h4"
          role="search"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-13pwdle-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
          <label
            class="css-inngth"
            for="main-search"
          >
            Search about documents
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-15tv2wk-MuiInputBase-root-MuiInput-root"
            required=""
          >
            <input
              aria-describedby="main-search-helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="main-search"
              name="search"
              placeholder="Search by Name..."
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </form>
      <div
        class="MuiBox-root css-ro3nde"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-quq592-MuiAvatar-root"
        >
          <img
            alt="Mr. Director avatar"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="assets"
          />
        </div>
        <div
          class=" MuiBox-root css-yi99sy"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-14tr2nh-MuiTypography-root"
          >
            Mr. Director
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-j1f0n7-MuiTypography-root"
          >
            Managing Director
          </p>
        </div>
      </div>
    </div>
  </header>
  <main
    class="MuiBox-root css-z2c4ih"
  >
    <div
      class="MuiBox-root css-0"
      data-testid="contacts"
    >
      <div
        class="MuiBox-root css-bh20kw"
      >
        <h1
          class="MuiTypography-root MuiTypography-h3 css-qcm841-MuiTypography-root"
        >
          Total Contacts
        </h1>
      </div>
      <div
        class="MuiBox-root css-0"
      />
    </div>
  </main>
</div>
`;

exports[`HomePage tests is a snapshot with a screen width greater than 900px 1`] = `
<div
  class="MuiBox-root css-1svsvwq"
  data-testid="home-page"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-tfubrb-MuiPaper-root-MuiAppBar-root"
    data-testid="header"
  >
    <div
      class="MuiBox-root css-sdx3s9"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ubps8a-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        open navigation menu
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-112rk81-MuiTypography-root-MuiLink-root"
        href="/contacts"
      >
        <img
          alt="logo"
          class="css-12pywbb"
          src="assets"
        />
      </a>
      <form
        class="css-1pho46g"
        data-testid="search-form"
        novalidate=""
      >
        <div
          class="MuiBox-root css-19sk4h4"
          role="search"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-13pwdle-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
          <label
            class="css-inngth"
            for="main-search"
          >
            Search about documents
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-15tv2wk-MuiInputBase-root-MuiInput-root"
            required=""
          >
            <input
              aria-describedby="main-search-helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="main-search"
              name="search"
              placeholder="Search by Name..."
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </form>
      <div
        class="MuiBox-root css-ro3nde"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-quq592-MuiAvatar-root"
        >
          <img
            alt="Mr. Director avatar"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="assets"
          />
        </div>
        <div
          class="open MuiBox-root css-yi99sy"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-14tr2nh-MuiTypography-root"
          >
            Mr. Director
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-j1f0n7-MuiTypography-root"
          >
            Managing Director
          </p>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-13g7zun-MuiDrawer-docked"
    data-testid="navigation"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiBox-root css-1ndpvz5"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-h74647-MuiTypography-root-MuiLink-root"
          href="/contacts"
        >
          <img
            alt="logo"
            class="css-12pywbb"
            src="assets"
          />
        </a>
      </div>
      <nav
        class="MuiBox-root css-10klw3m"
      >
        <ul
          class="MuiList-root MuiList-padding css-t39185-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
          >
            <a
              aria-current="page"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root active"
              href="/contacts"
            >
              <span
                class="MuiBox-root css-tyy0l3"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ContactsIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 0H4v2h16V0zM4 24h16v-2H4v2zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-8 2.75c1.24 0 2.25 1.01 2.25 2.25s-1.01 2.25-2.25 2.25S9.75 10.24 9.75 9 10.76 6.75 12 6.75zM17 17H7v-1.5c0-1.67 3.33-2.5 5-2.5s5 .83 5 2.5V17z"
                  />
                </svg>
              </span>
              Total Contacts
            </a>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
              href="/calendar"
            >
              <span
                class="MuiBox-root css-tyy0l3"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DateRangeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                  />
                </svg>
              </span>
              Calendar
            </a>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
              href="/project-report"
            >
              <span
                class="MuiBox-root css-tyy0l3"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DataUsageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53C16.17 17.98 14.21 19 12 19z"
                  />
                </svg>
              </span>
              Project Report
            </a>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-req3s6-MuiTypography-root-MuiLink-root"
              href="/login"
            >
              <span
                class="MuiBox-root css-tyy0l3"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LogoutIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                  />
                </svg>
              </span>
              Log out
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <main
    class="MuiBox-root css-1rbjnq0"
  >
    <div
      class="MuiBox-root css-0"
      data-testid="contacts"
    >
      <div
        class="MuiBox-root css-bh20kw"
      >
        <h1
          class="MuiTypography-root MuiTypography-h3 css-qcm841-MuiTypography-root"
        >
          Total Contacts
        </h1>
      </div>
      <div
        class="MuiBox-root css-0"
      />
    </div>
  </main>
</div>
`;
